### PR TITLE
eos-extra-pwa: Include Epithany Compatible Web Apps

### PR DESCRIPTION
**Pull request by MichaelDrydenEndless**
Added sites from Epithany Compatible Web Apps Spreadsheet

https://phabricator.endlessm.com/T34583

### CODE REVIEW

**Comment by MichaelDrydenEndless** [comment]
I tried running this through the pwa-metainfo-generator but got a No Manifest error that prevented me from generating the XML files.

**Review by dbnicholson — CHANGES_REQUESTED** [review]
Please generate and commit the appstream XML so we can see what the final output looks like.

**Comment by dbnicholson** [comment]
> I tried running this through the pwa-metainfo-generator but got a "No Manifest error" starting with the sites added here that prevented me from generating the XML files.

Which site was causing that?

**Comment by dbnicholson** [comment]
> Here is the list of which ones do/don't have manifests: No Manifest:

OK, I think we'll have to change the generator to not require a manifest. When this was initially conceived, the generator was a simple manifest to appstream converter. Now we're also trying to gather the data from other places and the manifest is just one source.

Here's what I'd like you to do. Remove `ManifestNotFoundException`. Where it was used, instead return an empty dictionary like `return {}`. Wherever the code was looking up an entry in the manifest like `manifest['something']`, change it to `manifest.get('something')`. This will return `None` if there's no `something` entry in the manifest dictionary (which will be the case if there's no manifest at all). If `None` isn't appropriate, you can use a different default fallback the 2nd parameter to `get`. Like, `manifest.get('icons', [])` might be appropriate since the code all treats the icons as a list.

> Additionally, I found that there were a number of sites that threw an error that prevented me from retrieving the manifest 403 Client Error: Forbidden for url:
> 
> * https://www.canva.com/
> * https://www.imdb.com/
> * https://accuweather.com/
> * https://quizlet.com/
> * https://cookpad.com/us

Ugh. These are all sites that appear to either block specific user agents in the request or require JavaScript to pass a captcha. We might be able to put something in the generator to deal with that, but I don't think it's really worth the effort.

As an FYI, in #38 I'm removing the YAML setup and the generator will just provide a getting started template and then you're expected to directly edit the appstream until it's appropriate. In that case, we just won't run the generator for these sites at all and the appstream will be built from scratch.

For the moment, just skip these and we'll come back to them.

**Comment by MichaelDrydenEndless** [comment]
Some of the committed files had conflicts but I think I cleared all those up and got it all current.

**Comment by dbnicholson** [comment]
This is looking pretty good, but now there are a bunch of conflicts since #38 and #40 were merged. An important thing to note is that #38 got rid of the YAML file. Instead, you'll generate the metainfo XML files and then edit them directly.

In GitHub Desktop, you can go to Branch -> Rebase current branch and then choose master. This will hit conflicts that you have to resolve. The rebase dialog will prompt you to address them, but it can be tricky to understand what's going on. In the case of the YAML file, you would say that you want to keep the deleted version from master. For the metainfo files, you have to open them in an editor and fix up the issues. https://youtu.be/OXtdxHTh2oY is a good description of what's happening. When all the issues are resolved, GitHub Desktop will tell you you can continue.

Since this one was pretty difficult, I did the rebase myself and pushed it to [T34583-dbn](https://github.com/endlessm/gnome-software-data/tree/T34583-dbn). I did 2 other things that might be interesting there:

* I [reordered](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-commits/reordering-commits) the script fix so it came first in the series.
* I split the metainfo changes into 2 commits - the first just fixes the existing GNOME PWA ones, and the second adds the new web apps. I did Undo commit in GitHub Desktop and then unchecked the new files (actually I did it on the CLI by not adding the new files for the first commit, but I think it's the same idea).

If you want to just use my branch, you can do `git reset --hard origin/T34583-dbn` and `git push --force origin T34583`.

File: app-info/metainfo/org.gnome.Software.WebApp_092f50a2d722b80d5fad80fefb89559d0c56b3e2.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="web-application">
+  <id>org.gnome.Software.WebApp_092f50a2d722b80d5fad80fefb89559d0c56b3e2.desktop</id>
+  <name />
```
Comment (dbnicholson) [inline app-info/metainfo/org.gnome.Software.WebApp_092f50a2d722b80d5fad80fefb89559d0c56b3e2.metainfo.xml:4]: This needs a name.

File: app-info/metainfo/org.gnome.Software.WebApp_092f50a2d722b80d5fad80fefb89559d0c56b3e2.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  <launchable type="url">https://archive.org/</launchable>
+  <url type="homepage">https://archive.org/</url>
+  <summary>Providing Universal Access to All Knowledge</summary>
+  <description>Internet Archive is a non-profit digital library offering free universal access to books, movies &amp; music, as well as 624 billion archived web pages.</description>
+  <project_license>GPL-2.0-or-later</project_license>
```
Comment (dbnicholson) [inline app-info/metainfo/org.gnome.Software.WebApp_092f50a2d722b80d5fad80fefb89559d0c56b3e2.metainfo.xml:9]: All the `<description>` tags need to have markup. Sorry I didn't catch this earlier. You can just take each one and add a `<p>` at the beginning and `</p>` at the end to wrap it in a paragraph.

File: app-info/metainfo/org.gnome.Software.WebApp_3a023ac6fc65c87ba5c88c1a1bb07edef7c6b8c4.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="web-application">
+  <id>org.gnome.Software.WebApp_3a023ac6fc65c87ba5c88c1a1bb07edef7c6b8c4.desktop</id>
+  <name />
```
Comment (dbnicholson) [inline app-info/metainfo/org.gnome.Software.WebApp_3a023ac6fc65c87ba5c88c1a1bb07edef7c6b8c4.metainfo.xml:4]: Missing name.

File: app-info/metainfo/org.gnome.Software.WebApp_506b15e469a34ac26f7b3235568af3c0cc5b3164.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="web-application">
+  <id>org.gnome.Software.WebApp_506b15e469a34ac26f7b3235568af3c0cc5b3164.desktop</id>
+  <name />
```
Comment (dbnicholson) [inline app-info/metainfo/org.gnome.Software.WebApp_506b15e469a34ac26f7b3235568af3c0cc5b3164.metainfo.xml:4]: Missing name.

File: app-info/metainfo/org.gnome.Software.WebApp_52b732dd06d99298b6e9bfccc17e25e0ac8b2bcb.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="web-application">
+  <id>org.gnome.Software.WebApp_52b732dd06d99298b6e9bfccc17e25e0ac8b2bcb.desktop</id>
+  <name />
```
Comment (dbnicholson) [inline app-info/metainfo/org.gnome.Software.WebApp_52b732dd06d99298b6e9bfccc17e25e0ac8b2bcb.metainfo.xml:4]: Missing name.

File: app-info/metainfo/org.gnome.Software.WebApp_7236bcfcc34bbdb54ff31a49ec40cdc2fd50c6e0.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="web-application">
+  <id>org.gnome.Software.WebApp_7236bcfcc34bbdb54ff31a49ec40cdc2fd50c6e0.desktop</id>
+  <name />
```
Comment (dbnicholson) [inline app-info/metainfo/org.gnome.Software.WebApp_7236bcfcc34bbdb54ff31a49ec40cdc2fd50c6e0.metainfo.xml:4]: Missing name.

File: app-info/metainfo/org.gnome.Software.WebApp_9e300f8f00666f4790eb7ed1b9e1c92c598759cb.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,39 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="web-application">
+  <id>org.gnome.Software.WebApp_9e300f8f00666f4790eb7ed1b9e1c92c598759cb.desktop</id>
+  <name />
```
Comment (dbnicholson) [inline app-info/metainfo/org.gnome.Software.WebApp_9e300f8f00666f4790eb7ed1b9e1c92c598759cb.metainfo.xml:4]: Missing name.

**Review by dbnicholson — CHANGES_REQUESTED** [review]
Here's a few comments on the metainfo files. Since the YAML file has been removed, please add the URL for each new app to `app-info/eos-extra-pwa.txt` so we have an easy record for each PWA we're managing.

**Comment by MichaelDrydenEndless** [comment]
I've added the URLs and fixed the metainfo xml files.

**Review by dbnicholson — APPROVED** [review]
I think this is good enough for now. All the apps need screenshots and the appstream catalog (the `eos-extra.xml` file) needs to be regenerated to actually use this data.

**Pull request merged**